jaeger-thrift/src/main/java/io/jaegertracing/thrift/internal/senders/ThriftSender.java

```diff
@@ -54,9 +54,9 @@ public int append(JaegerSpan span) throws SenderException {
 
     io.jaegertracing.thriftjava.Span thriftSpan = JaegerThriftSpanConverter.convertSpan(span);
     int spanSize = calculateSpanSize(thriftSpan);
-    if (spanSize > getMaxSpanBytes()) {
+    if ((processBytesSize + spanSize) > getMaxSpanBytes()) {
       throw new SenderException(String.format("ThriftSender received a span that was too large, size = %d, max = %d",
-          spanSize, getMaxSpanBytes()), null, 1);
+          (processBytesSize + spanSize), getMaxSpanBytes()), null, 1);
```

Comment:
I feel like this fix is not completely accurate. Look at the statements below, which again compare accumulated buffer size and getMaxSpanBytes(). I may be wrong, but it seems that the issue is with `getMaxSpanBytes()` - from its name it seems to mean "total space allocated for _span bytes only_, excluding process bytes", but the way it is calculated in the constructor is different, it does not include process bytes.

Reply:
@yurishkuro If you're referring to the if statements at line 63-69, when `process size + span size` exceeds the maximum byte size the if statements are bypassed (since `byteBufferSize` would be greater than `getMaxSpanBytes()`) then an exception is thrown at line 73 (`flush()`). When this happens `ThriftUdpTransport` is left in an unrecoverable state, as I mentioned in the PR description. You can reproduce this by running the test cases.

Reply:
Or do you mean the fix should be to modify getMaxSpanBytes() to return the space excluding the process bytes?

Reply:
I am not sure which one is best, it's just that the logic is getting very confusing. `getMaxSpanBytes()`, despite its name, returns the capacity of _both_ the spans and the process. Same for `byteBufferSize` variable - we have to keep remembering to add `processBytesSize`. For example, in L73 we call `flush()`, which assigns `byteBufferSize = byteBufferSize = processBytesSize;`, but then in L80 we add `processBytesSize` again.

Perhaps we should:
  * rename getMaxSpanBytes to getMaxBatchBytes in the base class and add comment that the size includes `process + spans`
  * implement getMaxSpanBytes just in this class as `getMaxBatchBytes() - processBytesSize´

Reply:
That makes total sense. Will update soon.